### ISSUE TITLE
GetType() returns the wrong type due to lack of element type information for arrays

### ISSUE THREAD

**Issue by TriEdgeAI** [bridgedotnet/Bridge#2056]
### Expected

```
System.Int32[]
System.User[]
System.Array
```

### Actual

```
Array
Array
Array
```

### Steps To Reproduce

http://deck.net/690c6a3cb1929e3cc16244417e983bc9

```cs
public class Program
{
    public class User
    {
        //...
    }
    
    public static void Main()
    {
        Console.WriteLine(typeof(int[]).FullName); 
        Console.WriteLine(typeof(User[]).FullName); 
        Console.WriteLine(typeof(Array).FullName);
    }
}
```

**Comment by michaelcheers** [comment]
Related to #1346.

**Comment by michaelcheers** [comment]
Duplicate of #1347, point 4.

**Comment by LeonidVeriga** [comment]
Yes, duplicate of #1347, point 4 - update https://github.com/bridgedotnet/Bridge/issues/1347#issuecomment-263328892 with a progress when fixed.

**Comment by geoffreymcgill** [comment]
Related to #2051 and #2052.

**Comment by geoffreymcgill** [comment]
This feature has been implemented and a pull request has been submitted. The feature is expected to be included in the Bridge 15.7 release.
